### PR TITLE
added board systems (allowing for board renaming)

### LINKED ISSUE #80: Board Options
showing/allowing for the editing of board options

### PR DESCRIPTION
**Pull request by weihanli101**
This closes #80

### CODE REVIEW

File: src/app/components/board-settings-dialog/board-settings-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<mat-toolbar class="settings-title" >Board Settings</mat-toolbar>
```
Comment (thechausenone) [inline src/app/components/board-settings-dialog/board-settings-dialog.component.html]: perhaps "Board Details"?

File: src/app/components/board-settings-dialog/board-settings-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+<div mat-dialog-content>
+  <mat-form-field>
+      <input matInput type="text" placeholder="Name" [(ngModel)]="boardtitle"/>
```
Comment (thechausenone) [inline src/app/components/board-settings-dialog/board-settings-dialog.component.html]: Could you add an option to change the icon as well?

File: src/app/components/board-settings-dialog/board-settings-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import { Subscription } from "rxjs/Subscription";
+
+@Component({
+  selector: 'app-board-settings-dialog',
+  templateUrl: './board-settings-dialog.component.html',
+  styleUrls: ['./board-settings-dialog.component.scss']
+})
+export class BoardSettingsDialogComponent {
+
+  private _selectedFile: string;
```
Comment (thechausenone) [inline src/app/components/board-settings-dialog/board-settings-dialog.component.ts]: I think the "selectedFile" variable was only for the windows dialog, so we probably don't need it here

File: src/app/components/board-settings-dialog/board-settings-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    var boardName = this.activeBoard.title;
+    this._dataService.deleteBoard(boardName);
+    this._dataService.getBoards();
+  }
+
+  onChange(event){
+    this._selectedFile = event.srcElement.files[0].path;
+  }
+
+  renameBoard(){
```
Comment (thechausenone) [inline src/app/components/board-settings-dialog/board-settings-dialog.component.ts]: when #81 is merged, you're gonna have to add a function in StateManagerService for board renaming

File: src/app/components/navbar/navbar.component.html [hunk 5]
```diff
@@ -29,7 +29,7 @@
 
     <mat-menu #menu="matMenu">
       <button mat-menu-item (click) = "mapWindowsToDesktop()">Map windows</button>
-      <button mat-menu-item (click) = "deleteBoard()">Delete</button>
+      <button mat-menu-item (click) = "boardSettings()">Board Settings</button>
```
Comment (thechausenone) [inline src/app/components/navbar/navbar.component.html]: "Edit Board"?

File: src/app/components/board-settings-dialog/board-settings-dialog.component.html (hunk 1 above)
Comment (weihanli101) [reply src/app/components/board-settings-dialog/board-settings-dialog.component.html]: Probably just keep it as settings. Editor is kinda of subjective. Settings gets the point across and should we expand on this project later we can different features to it.

File: src/app/components/navbar/navbar.component.html (hunk 5 above)
Comment (weihanli101) [reply src/app/components/navbar/navbar.component.html]: Going to leave as settings as mention above.

**Pull request merged**